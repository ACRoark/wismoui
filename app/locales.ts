export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
