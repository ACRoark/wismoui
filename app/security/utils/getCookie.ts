const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
