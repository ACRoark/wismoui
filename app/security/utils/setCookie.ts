interface ICookieOptions {
  days: number;
  path: string;
}

const setCookie = (name: string, value: string, options: ICookieOptions): void => {
  const expires = new Date(Date.now() + options.days * 864e5).toUTCString();

  document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=${options.path}`;
};

export default setCookie;
