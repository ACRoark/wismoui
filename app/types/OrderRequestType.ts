export const OrderRequestTypes = ['MIMO_START', 'MIMO_STOP'] as const;

type OrderRequestType = typeof OrderRequestTypes[number];

export default OrderRequestType;
