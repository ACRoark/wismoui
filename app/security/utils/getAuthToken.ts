import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
