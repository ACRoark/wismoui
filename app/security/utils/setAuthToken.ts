import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
