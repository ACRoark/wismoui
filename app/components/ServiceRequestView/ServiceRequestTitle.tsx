import React, { FC, ReactElement } from 'react';
import { FormattedMessage } from 'react-intl';

import OrderRequestType from 'types/OrderRequestType';
import './index.less';
import messages from './messages';

interface IServiceRequestTitle {
  orderRequestType: OrderRequestType;
}

const ServiceRequestTitle: FC<IServiceRequestTitle> = (props: IServiceRequestTitle): ReactElement => {
  const { orderRequestType } = props;

  return (
    <div className="dte-wismo-service-request-title">
      {orderRequestType === 'MIMO_START' ? (
        <FormattedMessage {...messages.startTitle} />
      ) : (
        <FormattedMessage {...messages.stopTitle} />
      )}
    </div>
  );
};

export default ServiceRequestTitle;
