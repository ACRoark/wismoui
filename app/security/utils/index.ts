import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
