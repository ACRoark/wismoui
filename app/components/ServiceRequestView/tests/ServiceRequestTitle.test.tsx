import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';

import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import OrderRequestType from 'types/OrderRequestType';

import ServiceRequestTitle from '../ServiceRequestTitle';

const orderRequests: OrderRequestType[] = ['MIMO_START', 'MIMO_STOP'];

describe('ServiceRequestTitle', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    orderRequests.forEach((orderRequest: string): void => {
      it(`should not log errors in console when rendering a ${orderRequest} order in ${language}`, (): void => {
        const spy = jest.spyOn(global.console, 'error');

        renderWithIntl(<ServiceRequestTitle orderRequestType={orderRequest as OrderRequestType} />, language);

        expect(spy).not.toHaveBeenCalled();
      });

      it(`should render the correct start title when the language is ${language} and orderRequestType is ${orderRequest}`, (): void => {
        const tree = createSnapshotWithIntl(<ServiceRequestTitle orderRequestType={orderRequest as OrderRequestType} />, language);

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
