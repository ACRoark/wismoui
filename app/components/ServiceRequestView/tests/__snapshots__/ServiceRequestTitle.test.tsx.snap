// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceRequestTitle should render the correct start title when the language is en and orderRequestType is MIMO_START 1`] = `
<div
  className="dte-wismo-service-request-title"
>
  Start Service
</div>
`;

exports[`ServiceRequestTitle should render the correct start title when the language is en and orderRequestType is MIMO_STOP 1`] = `
<div
  className="dte-wismo-service-request-title"
>
  Stop Service
</div>
`;

exports[`ServiceRequestTitle should render the correct start title when the language is es and orderRequestType is MIMO_START 1`] = `
<div
  className="dte-wismo-service-request-title"
>
  Comienza el servicio
</div>
`;

exports[`ServiceRequestTitle should render the correct start title when the language is es and orderRequestType is MIMO_STOP 1`] = `
<div
  className="dte-wismo-service-request-title"
>
  Parada de servicio
</div>
`;
